src/heap/heap.ts

```diff
@@ -0,0 +1,76 @@
+export default class Heap<T> {
+  protected memory: T[] = [];
+
+  constructor(protected CompareFunction: (a: T, b: T) => number) {
+  }
+
+  public insert(element : T): void {
+    this.memory.push(element);
+  }
+
+  protected up(placeInArray: number, typeOfHeap: 'min' | 'max'): void {
+    const elementToUp = this.memory[placeInArray];
+    const placeInArrayOfParentNode = this.parent(placeInArray);
+    const isRoot = this.CompareFunction(elementToUp, this.memory[0]) === 0;
+    const comparer = typeOfHeap === 'min'
```

Comment:
comparer is not a word, it could be comparator. But still the name of that variable should be isMin